crates/pink/pink-extension-runtime/src/lib.rs

```diff
@@ -64,15 +64,26 @@ impl<T: PinkRuntimeEnv, E: From<&'static str>> PinkExtBackend for DefaultPinkExt
             headers.insert(key, value);
         }
 
-        let mut response = client
+        let result = client
             .request(method, url)
             .headers(headers)
             .body(request.body)
-            .send()
-            .map_err(|err| {
+            .send();
+
+        let mut response = match result {
+            Ok(response) => response,
+            Err(err) => {
+                // If there is somthing wrong with the network, we can not inspect the reason too
```

Comment:
Why don't we pass it to the contract?

Reply:
Don't want to break the API and the guest is already required to check the status_code. Also, the original error is not encodable to pass to the guest.

Reply:
OK. Just realized the error is not actually a HTTP status, but outside the http session. However, 408 means [the connection is closed by the remote server](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/408). Do you think it's better to use some non-standard HTTP error code?

Reply:
https://en.wikipedia.org/wiki/List_of_HTTP_status_codes#Unofficial_codes
Doesn't match exactly, but find two similar meaning codes in Cloudflare's extension. Let's partially follow it.

Reply:
Cool